lib/class-wp-theme-json-gutenberg.php

```diff
@@ -289,6 +310,28 @@ public function __construct( $theme_json = array() ) {
 		$valid_block_names   = array_keys( self::get_blocks_metadata() );
 		$valid_element_names = array_keys( self::ELEMENTS );
 		$this->theme_json    = self::sanitize( $theme_json, $valid_block_names, $valid_element_names );
+
+		// Internally, presets are keyed by origin.
```

Comment:
Would it make sense instead of having this loop to just call merge inside the constructor?

Reply:
The merge was changed and simplified so my comment is no longer valid.

Reply:
mmm, not sure how, they do different things:

- the `constructor` takes an array, and presets are unkeyed (data that comes from core's or theme's `theme.json`, data stored in the database for user)
- the `merge` takes a `WP_Theme_JSON` object, which holds presets by origin